.github/workflows/_check-user.yml

```diff
@@ -0,0 +1,72 @@
+name: Check whether the workflow owner can use ARC runners
+
+on:
+  workflow_call:
+    inputs:
+      user_name:
+        required: true
+        type: string
+        description: The name of the workflow owner.
+      curr_branch:
+        required: true
+        type: string
+        description: Current branch.
+
+    outputs:
+      workflow-type:
+        description: Type of runners to use
+        value: ${{ jobs.check-user.outputs.workflow-type }}
+
+jobs:
+  check-user:
+    runs-on: linux.4xlarge
+    outputs:
+      workflow-type: ${{ steps.set-condition.outputs.workflow-type }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Check if user is in arc_users.yml
+        id: set-condition
+        run: |
+          curr_branch="${{ inputs.curr_branch }}"
+          echo "Current branch is '$curr_branch'"
+
+          if [[ "$curr_branch" == "nightly" || "$curr_branch" == "main" || "$curr_branch" == "release/"* || "$curr_branch" == "landchecks/"* ]]; then
+            echo "workflow-type=label" >> "$GITHUB_OUTPUT"
+          else
+            REPO_OWNER="pytorch"
+            REPO_NAME="test-infra"
+            ISSUE_NUMBER="5132"
+
+            COMMENTS=$(curl -H "Authorization: token $GITHUB_TOKEN" "https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/issues/$ISSUE_NUMBER/comments")
```

Comment:
Let's do two things for safety
1. Make this action have a safe fall back (to classic-infra) only if either the curl request or parsing fails
2. Update the issue to point to this action so that future maintainers can easily find where it's used

Reply:
1. I updated the code so the fallback policy always returns "`label`".
2. The [issue number](https://github.com/pytorch/pytorch/pull/124721/files#diff-5900a66fe6f5f773249ace156f682e6f2ef3618c6f8b536438e2519c2240eb53R17) is an input parameter. Thus, it should be easy to define and configure.